src/test/java/com/google/graphgeckos/dashboard/api/DashboardControllerTest.java

```diff
@@ -0,0 +1,165 @@
+
+  /* Builders logs components. */
+  private final String LOG_TYPE_1 = "type 1";
+  private final String LOG_TYPE_2 = "type 2";
+  private final String LOG_TYPE_3 = "type 3";
+  private final String LOG_LINK_1 = "link 1";
+  private final String LOG_LINK_2 = "link 2";
+  private final String LOG_LINK_3 = "link 3";
+
+  /* Builders logs. */
+  private final List<Log> LOGS_A = Arrays.asList(new Log(LOG_TYPE_1, LOG_LINK_1),
+                                               new Log(LOG_TYPE_2, LOG_LINK_2));
+  private final List<Log> LOGS_B = Collections.singletonList(new Log(LOG_TYPE_3, LOG_LINK_3));
+
+  private final BuildBotData BUILDER_A = new BuildBotData(
+                                        COMMIT_HASH, NAME_A, LOGS_A, BuilderStatus.valueOf(STATUS_A));
+  private final BuildBotData BUILDER_B = new BuildBotData(
+                                         COMMIT_HASH, NAME_B, LOGS_B, BuilderStatus.valueOf(STATUS_B));
+
+  private final BuildInfo BUILD_INFO = new BuildInfo(INITIAL_GITHUB_DATA);
+
+
+  /* Illegal number of entries value. Revision must be >= 0. */
+  private final int MINUS_ONE_ENTRIES = -1;
+
+  /* Valid number of entries and offset values. */
+  private final int ZERO_ENTRIES = 0;
+  private final int ONE_ENTRIES = 1;
+  private final int OFFSET_ZERO = 0;
+
+  /** Enables Mockito.*/
+  @Before
+  public void init() {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @Before
+  public void setUpBuildInfo() {
+    BUILD_INFO.addBuilder(BUILDER_A);
+    BUILD_INFO.addBuilder(BUILDER_B);
+  }
+
+  /**
+   * Given: number of the entries to return = 0, any offset >= 0
+   * Expected behaviour: empty JSON list, Response Status 200 (OK).
+   */
+  @Test
+  public void whenGivenZeroRevisionsOffsetZeroReturnsEmptyJSONArray() throws Exception {
+
+    given(datastoreRepository.getLastRevisionEntries(ZERO_ENTRIES, OFFSET_ZERO))
+      .willReturn(Collections.emptyList());
+
+    ResultActions result = mvc.perform(
+      MockMvcRequestBuilders.get("/builders/number={number}/offset={offset}",
+        ZERO_ENTRIES, OFFSET_ZERO)
+        .accept(MediaType.APPLICATION_JSON));
+
+    String emptyJsonList = "[]";
+    result.andExpect(status().isOk())
+          .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+          .andExpect(content().string(emptyJsonList));
+  }
+
+  /**
+   * Should return the JSON list of length.
+   *
+   * Given: number of entries to return >= 0; offset b, b >= 0.
+   * Expected behaviour: JSON list of length = number
+   * of entries to return, more about offset and number at
+   * {@link com.google.graphgeckos.dashboard.storage.DataRepository::getLastRevision}.
+   * Returned list contains JSON representations of the BuildBotInfo instances.
+   * Information about this JSON representations could be found at
+   * {@link com.google.graphgeckos.dashboard.datatypes.BuildInfo}.
+   */
+  @Test
+  public void whenGivenOneRevisionOffsetZeroReturnsOneBuildInfoJson() throws Exception {
+
```

Comment:
nit: remove blank line at the start of the code

Reply:
Same as above